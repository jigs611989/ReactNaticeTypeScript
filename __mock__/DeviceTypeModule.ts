// import React from 'react';
const DeviceTypeModule = {
  // tslint:disable-next-line:no-empty
  checkDeviceType: () => {},
};
export const checkDeviceType = DeviceTypeModule.checkDeviceType;
export default DeviceTypeModule;
