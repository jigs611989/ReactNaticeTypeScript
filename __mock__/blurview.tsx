import React from 'react';
import { View } from 'react-native';
const BlurView = () => <View />;
export { BlurView };
