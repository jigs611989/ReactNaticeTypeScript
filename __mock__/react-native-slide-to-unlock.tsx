import React from 'react';
import { View } from 'react-native';
const Slider = () => <View />;
export default Slider;
