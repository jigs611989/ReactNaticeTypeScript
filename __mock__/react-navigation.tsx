const useNavigation = () => {
  return {
    navigate: () => () => {},
  };
};

export { useNavigation };
