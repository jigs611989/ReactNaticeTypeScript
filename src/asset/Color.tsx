const Colors = {
  primary: '#f7a934',
  black: '#0B121E',
  white: '#FFFFFF',
  placeholder: '#747577',
  blueGreyBackground: 'rgba(11, 18, 30, 0.02)',
  blueGreyBorder: 'rgba(11, 18, 30, 0.1)',
  blueGrey: 'rgb(11, 18, 30)',
  darkText: '#101313',
}

export default Colors;
