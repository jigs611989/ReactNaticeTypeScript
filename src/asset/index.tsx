export { default as Color } from './Color';
export { default as AppStyle } from './AppStyle';
export { default as String } from './String';
export { default as Images } from './image';
