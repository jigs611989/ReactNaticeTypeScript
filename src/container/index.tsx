export { default as WelcomeScreen } from './WelcomeScreen';
export { default as HomeScreen } from './HomeScreen';
export { default as ButtonOptionScreen } from './ButtonOptionScreen';
