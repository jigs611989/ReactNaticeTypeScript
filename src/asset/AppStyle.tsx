import { StyleSheet } from 'react-native';
import Color from './Color';
const AppStyle = StyleSheet.create({
  rootContainer: {
    flex: 1,
    backgroundColor: Color.black,
  },
  buttonStyle: {
    backgroundColor: Color.primary,
  },
});

export default AppStyle;
