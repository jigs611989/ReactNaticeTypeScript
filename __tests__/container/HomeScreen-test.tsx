import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { HomeScreen } from '../../src/container';
import { render, fireEvent } from '@testing-library/react-native';
import { String } from '../../src/asset';
import { mockedNavigate } from '../../__mock__/setup';

describe('HomeScreen', () => {

  it('render correctly', () => {
    
    const mockStore = configureStore();
    const initialState = { name: '' };
    const store = mockStore(initialState);
    const { toJSON, getByText } = render(
      <Provider store={store}>
        <HomeScreen />
      </Provider>,
    );

    // check button click event
    const buttonOption = getByText(String.Buttons.buttonOption);
    fireEvent.press(buttonOption);
    expect(mockedNavigate).toHaveBeenCalledTimes(1);

    // should match the snapshot
    expect(toJSON()).toMatchSnapshot();
  });

});


