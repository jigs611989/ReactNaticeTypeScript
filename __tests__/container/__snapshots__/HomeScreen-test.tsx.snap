// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#0B121E",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 24,
            "lineHeight": 28,
            "margin": 16,
            "marginBottom": 2,
          },
          Object {
            "color": "#FFFFFF",
            "fontSize": 24,
            "lineHeight": 28,
            "margin": 16,
            "marginBottom": 2,
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/asset/image/ReactLoGo.png",
      }
    }
    style={
      Object {
        "height": 300,
        "resizeMode": "contain",
        "width": 300,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f7a934",
        "borderRadius": 10,
        "height": 50,
        "justifyContent": "center",
        "margin": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Button Option Screen
    </Text>
  </View>
</RCTSafeAreaView>
`;
